src/commands/plugins.rs

```diff
@@ -242,11 +243,72 @@ impl Upgrade {
 
         if self.all {
             self.upgrade_all(manifests_dir).await
+        } else if self.name.is_none()
+            && self.local_manifest_src.is_none()
+            && self.remote_manifest_src.is_none()
+        {
+            // Default behavior (multiselect)
+            self.upgrade_multiselect().await
         } else {
             self.upgrade_one().await
         }
     }
 
+    // Multiselect plugin upgrade experience
+    async fn upgrade_multiselect(self) -> Result<()> {
+        let catalogue_plugins = list_catalogue_plugins().await?;
+        let installed_plugins = list_installed_plugins()?;
+
+        let installed_in_catalogue: Vec<_> = installed_plugins
+            .into_iter()
+            .filter(|installed| {
+                catalogue_plugins.iter().any(|catalogue| {
+                    installed.manifest == catalogue.manifest
+                        && matches!(catalogue.compatibility, PluginCompatibility::Compatible)
+                })
+            })
+            .collect();
+
+        eprintln!(
+            "Select plugins to upgrade. Use Space to select/deselect and Enter to confirm selection."
+        );
+        let selected_indexes = match dialoguer::MultiSelect::new()
+            .items(&installed_in_catalogue)
+            .interact_opt()?
+        {
+            Some(indexes) => indexes,
+            None => return Ok(()),
+        };
+
+        let plugins_selected = elements_at(installed_in_catalogue, selected_indexes);
+
+        if plugins_selected.is_empty() {
+            eprintln!("No plugins selected");
+            return Ok(());
+        }
+
+        // Upgrade plugins selected
+        for plugin in plugins_selected {
+            let manager = PluginManager::try_default()?;
+            let manifest_location = ManifestLocation::PluginsRepository(PluginLookup::new(
+                &plugin.name,
+                Some(plugin.version.parse::<Version>()?),
```

Comment:
I am not sure this is quite right.  `installed_in_catalogue` contains the installed plugins _at their installed versions_.  This therefore appears to use the already-installed version as the version to upgrade to.  I may be misreading though?

Reply:
You are right. Based on my understanding of the code, it seems that to determine the version for an upgrade, I need to provide `'None'` as the version when calling `manifest_location()`. From the tests I've conducted, I believe I have it correct this time.